### PR TITLE
Dev fisher information

### PR DESCRIPTION
**Pull request by ojanik**
Added methods to the analysis object to calculate the Fisher information, covariance matrix (from the Cramer Rao Bound) and variances (again Cramer Rao bound). This assumes the Poisson likelihood.

### CODE REVIEW

File: pyForwardFolding/analysis.py [hunk 1]
```diff
@@ -71,6 +74,107 @@ def evaluate(
...
+            fisher_information = values @ values.T
+            fisher_dict[comp_name] = fisher_information
+
+        return jnp.sum(jnp.asarray([v for v in fisher_dict.values()]),axis=0)
+
+    def covariance(
+        self,
+        datasets: Dict[str, Dict[str, Union[np.ndarray, float]]],
+        parameter_values: Dict[str, Union[np.ndarray, float]],
+    ) -> jnp.ndarray:
```
Comment (chrhck) [inline pyForwardFolding/analysis.py:130]: Can you convert this into a static method so that one can use it with a precalculated FIM?

File: pyForwardFolding/analysis.py [hunk 2]
```diff
@@ -71,6 +74,107 @@ def evaluate(
...
+        """
+        fisher_information = self.fisher_information(datasets, parameter_values)
+        cov = jnp.linalg.inv(fisher_information)
+        return cov
+
+    def variance(
+        self,
+        datasets: Dict[str, Dict[str, Union[np.ndarray, float]]],
+        parameter_values: Dict[str, Union[np.ndarray, float]],
+    ) -> Dict[str, jnp.ndarray]:
```
Comment (chrhck) [inline pyForwardFolding/analysis.py:154]: see above

File: pyForwardFolding/analysis.py [hunk 3]
```diff
@@ -71,6 +74,107 @@ def evaluate(
 
         return output_dict, output_ssq_dict
     
+    def fisher_information(
```
Comment (chrhck) [inline pyForwardFolding/analysis.py:77]: Can you split this function into an instance method that calculates the gradients and a static method that calculates the FIM?

**Review by chrhck — CHANGES_REQUESTED** [review]

**Review by chrhck — APPROVED** [review]
